### PR TITLE
Badass CI

### PR DESCRIPTION
**Pull request by lovesegfault**
Reworks our CI, pending coveralls integration approval.

### CODE REVIEW

**Comment by lovesegfault** [comment]
Blocked on #2

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,12 +1,48 @@
...
-    name: Cargo Test
+  ci:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust:
+          - stable
+          - beta
+          - nightly
+          - 1.31.0  # MSRV
```
Comment (cpcloud) [inline .github/workflows/ci.yml:14]: `MSRV = Minimum Supported Rust Version`, for my own edification.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,12 +1,48 @@
...
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+
+      - name: Test
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+
+      - name: Format
```
Comment (cpcloud) [inline .github/workflows/ci.yml]: Should we run this check first? I assume that it'd fail faster than waiting for build/test first only to fail a format check.

File: .github/workflows/coverage.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+on: [push]
```
Comment (cpcloud) [inline .github/workflows/coverage.yml]: Any reason to leave the brackets here and not in `ci.yml`?

File: .github/workflows/coverage.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: Toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          override: true
+
+      - name: Clean
```
Comment (cpcloud) [inline .github/workflows/coverage.yml:18]: Why do we need to clean?

**Review by cpcloud — APPROVED** [review]
This looks really great, just a couple of questions.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (lovesegfault) [reply .github/workflows/ci.yml]: Good idea, done

File: .github/workflows/coverage.yml (hunk 3 above)
Comment (lovesegfault) [reply .github/workflows/coverage.yml]: Nope

File: .github/workflows/coverage.yml (hunk 4 above)
Comment (lovesegfault) [reply .github/workflows/coverage.yml:18]: Docs recommended it, so I added just to be safe

**Pull request merged**